lib/matplotlib/tests/test_backend_svg.py

```diff
@@ -371,6 +371,20 @@ def test_svg_default_metadata(monkeypatch):
     # Type
     assert 'StillImage' in buf
 
+    # Now make sure all the default metadata can be cleared.
+    with BytesIO() as fd:
+        fig.savefig(fd, format='svg', metadata={'Date': None, 'Creator': None,
+                                                'Format': None, 'Type': None})
+        buf = fd.getvalue().decode()
+
+    # Creator
+    assert mpl.__version__ not in buf
+    # Date
+    assert '1970-08-16' not in buf
+    # Format
+    assert 'image/svg+xml' not in buf
+    # Type
+    assert 'StillImage' not in buf
```

Comment:
I think these checks are good, but they depend on exactly these strings being written when the metadata isn't None. Is there any way to get the metadata from the saved file and more directly check that it doesn't have any of these keys in it?

Reply:
> but they depend on exactly these strings being written when the metadata isn't None.

That's exactly why I added them here instead of a new test. The few lines above are testing exactly that. The next test seems to be more involved and I can try to add another one modeling that in addition.

Also, shouldn't the label be `backend/svg` rather than `backend/pdf`?

Reply:
can you make sure that the same issue is not in pdf?  We try to keep them parallel as possible.

Reply:
By the same issue as in the clearing of the creation date? I believe it is working since I have been passing in `CreateDate` metadata to pdf saving function and it seems to be reproducible.

Reply:
There does not seem to be a test for that in the pdf backend, though I don't think that should be the scope of this PR = = .....

Reply:
See #17936 for a related issue with pdf

Reply:
Hmm, it seems that https://github.com/matplotlib/matplotlib/issues/17936 isn't really a bug? The document [linked](https://matplotlib.org/3.3.0/api/backend_pdf_api.html#matplotlib.backends.backend_pdf.PdfPages) seems pretty clear that,

1. `metadata=None` is the default value of the option
2. by default
    > Values have been predefined for 'Creator', 'Producer' and 'CreationDate'.

This also match the behavior documented and observed for svg backend.
I do think it might be good to allow using `metadata=None` to clear everything but that's a bigger behavior change and is breaking (ppl could be reasonably relying on that to restore the default). I agree though that such a change should be sync'd across all backends but this one is SVG specific. This isn't adding new features/changing the documented behavior (unlike https://github.com/matplotlib/matplotlib/issues/17936) but merely fixing an inconsistency between document and code.